code/modules/admin/verbs/mapping.dm

```diff
@@ -422,3 +422,99 @@ GLOBAL_VAR_INIT(say_disabled, FALSE)
 	popup.set_content(page_contents)
 	popup.open()
 
+/// Checks the atmos monitor, sensors, meters, vents, and injectors, and makes sure they dont overlap or do nothing.
+/client/proc/check_atmos_controls()
+	set name = "Check Atmos Chamber Devices"
+	set category = "Mapping"
+
+	if(SSticker.current_state == GAME_STATE_STARTUP)
+		to_chat(usr, "Game still loading, please run this again later!", confidential = TRUE)
+		return
+
+	message_admins(span_adminnotice("[key_name_admin(usr)] used the Test Atmos Controls debug command."))
+	log_admin("[key_name(usr)] used the Test Atmos Controls debug command.")
+
+	var/datum/radio_frequency/frequency = SSradio.return_frequency(FREQ_ATMOS_STORAGE)
+
+	/// broadcaster[id_tag] = machine
+	var/list/broadcasters = list()
+	/// listened_to[atmos_chamber_entry] = bool
+	/// TRUE means we have the corresponding id_tag being listened to by an atmos control computer.
+	var/list/listened_to = list()
+	/// broadcasted_to[id_tag[1]] = bool
+	/// TRUE means we have the corresponding id_tag being broadcasted to by a device, be it meter, sensors, etc.
+	var/list/broadcasted_to = list()
+
+	/// How many things dont fit the recognized subtypes.
+	var/invalid_machine = 0
+	/// How many things have invalid (messes with our delimiter) tags.
+	var/invalid_tag = 0
+	/// How many things have empty tags, invalids but much worse.
+	var/tagless = 0
+	/// How many things have duped id_tag.
+	var/duplicate_tag = 0
+	/// How many things are broadcasting without an atmos computer listening
+	var/not_heard = 0
+	/// How many atmos computers are listening to an empty tag.
+	var/not_told = 0
+
+	var/list/valid_device_types = list(
```

Comment:
deliberately not using typesof here?

Reply:
I run an istype check for this list, not an in. Dont think that'd be needed. Unless you want the type checking changed.

Reply:
This seems like the perfect usecase for a static typecache, would save the iteration and all too